### PR TITLE
Use k8sclient collection

### PR DESCRIPTION
**Pull request by xh3b4sd**
Getting https://github.com/giantswarm/operatorkit/pull/338 in.

### CODE REVIEW

File: service/service.go [hunk 1]
```diff
@@ -6,15 +6,14 @@ import (
 	"context"
 	"sync"
 
-	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
```
Comment (kopiczko) [inline service/service.go]: I'd prefer to import this as corev1alpha1 but I'm not big on this.

Comment (tuommaki) [reply service/service.go]: I think we occasionally have used `corev1` to alias `k8s.io/api/core/v1` - that gets potentially confusing quite quickly - WDYT?

Comment (kopiczko) [reply service/service.go]: It doesn't confuse me but I get your point.

**Comment by xh3b4sd** [comment]
Just tested cluster creation on `gauss`. Works out. Tomorrow we need to make a plan on how to proceed with everything we have.

**Comment by sslavic** [comment]
Some of the work from this and https://github.com/giantswarm/cert-operator/pull/228 can be extracted, merged to master and shipped in isolation from k8s+helm+cluster-api dependency updates - e.g. migration from deprecated to new internal library APIs, and migration from deprecated k8s API groups.

The rest should be merged into single consistently named branch/PR with all k8s+helm+cluster-api dependency updates and related changes. Once we get such branches for the whole stack (libs, operators, exporters, ... - and that's where consistent branch naming helps) with passing CI builds, some will need additional testing on clusters and once confident enough that it all might work we can start merging/shipping those branches.

**Pull request merged**